abnf/odata-abnf-construction-rules.txt

```diff
@@ -442,16 +442,34 @@ contextFragment = %s"Collection($ref)"
                 / singletonEntity [ navigation *( containmentNavigation ) [ "/" qualifiedEntityTypeName ] ] [ selectList ]
                 / qualifiedTypeName [ selectList ]
                 / entitySet ( %s"/$deletedEntity" / %s"/$link" / %s"/$deletedLink" )
-                / entitySet keyPredicate "/" contextPropertyPath [ selectList ]
+                / entitySet contextKeyPredicate "/" contextPropertyPath [ selectList ]
                 / entitySet [ selectList ] [ %s"/$entity" / %s"/$delta" ]
 
 entitySet = entitySetName *( containmentNavigation ) [ "/" qualifiedEntityTypeName ]
 
-containmentNavigation = keyPredicate [ "/" qualifiedEntityTypeName ] navigation
+containmentNavigation = contextKeyPredicate [ "/" qualifiedEntityTypeName ] navigation
 navigation            = *( "/" complexProperty [ "/" qualifiedComplexTypeName ] ) "/" navigationProperty
 
-selectList         = OPEN [ selectListItem *( COMMA selectListItem ) ] CLOSE
-selectListItem     = STAR ; all structural properties
+contextKeyPredicate = "(" ( contextKeyValue / contextKeyValuePair *( "," contextKeyValuePair ) ) ")"
+contextKeyValuePair = ( primitiveKeyProperty / keyPropertyAlias  ) EQ contextKeyValue
+contextKeyValue  = boolean
+                 / guid
+                 / dateTimeOffsetValue
+                 / date
+                 / timeOfDayValue
+                 / decimalValue
+                 / sbyteValue
+                 / byte
+                 / int16Value
+                 / int32Value
+                 / int64Value
+                 / stringValue
+                 / durationValue
+                 / enumValue
+stringValue = %x27 *( %x00-26 / %x27 %x27 / %x28-FF) %x27 ; SQUOTE = %x27

```

Comment:
```suggestion
stringValue = %x27 *( %x00-26 / %x27 %x27 / %x28-2E / "%2f" / %x30-FF) %x27 ; SQUOTE = %x27
```
According to TC 2024-02-21, `/` must be encoded in a stringValue.

Reply:
That was discussed as a possibility, no decision yet.

Reply:
TC 2024-05-15 agrees not to encode these characters.